pkg/api/v1/types.go

```diff
@@ -2165,6 +2165,14 @@ type PodProxyOptions struct {
 	Path string `json:"path,omitempty"`
 }
 
+// ServiceProxyOptions is the query options to a Service's proxy call.
+type ServiceProxyOptions struct {
```

Comment:
Yes, we can combine/resue. Of course, we should rename PodProxyOptions to SubresourceProxyOptions firstly, is this new name suitable? 
And could I open a new PR to combine/resue SubresourceProxyOptions? @lavalamp @liggitt